drivers/flash/flash_stm32f3x.c

```diff
@@ -129,6 +129,7 @@ static const struct flash_driver_api flash_stm32_api = {
 	.write = flash_stm32_write,
 	.erase = flash_stm32_erase,
 	.write_protection = flash_stm32_protection_set,
+	.write_block_size = 1,
```

Comment:
block size is half word on STM32F3

Reply:
@erwango so 2 bytes?

Reply:
@carlescufi: yes, 2. Otherwise it's good

Reply:
stm32f3 driver supports unaligned write, see: https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/flash/flash_stm32f3x.c#L93

Reply:
@erwango if it's good then please approve this change.

Reply:
@nvlsianpu: Well, looking into it, driver actually takes into account unaligned writes. Problem is that last leading byte of an unaligned write will be also written to 0x00 (data & 0x00FF operation on the remainder). And then this byte will no longer be writeable.
So write_block_size is indeed 1, but driver should be updated.

Reply:
@erwango So this driver has the bug.
I belief fixing of the driver is not in scope of this PR.

Reply:
@nvlsianpu: Sure. I'll take care of it

Reply:
@nvlsianpu: While correcting the issue, I had some testing and it appears that any attempt to write unaligned byte (of half-word) will end up in bus fault. So, even if driver allows to write single byte, STM32F3 flash does not allow unaligned writes.
To be transparent to the driver user, I think it would be better stating that write_block_size is 2

Reply:
Thx, I see - the driver allows to write any amount of data but it is limited to start writing for half-word aligned address.